evalml/pipelines/components/transformers/samplers/base_sampler.py

```diff
@@ -118,25 +118,26 @@ def __init__(
         sampler,
         sampling_ratio=0.25,
         sampling_ratio_dict=None,
-        k_neighbors=5,
+        k_neighbors_default=5,
         n_jobs=-1,
         random_seed=0,
-        **kwargs
+        **kwargs,
     ):
         """Initializes the oversampler component.
 
         Arguments:
             sampling_ratio (float): This is the goal ratio of the minority to majority class, with range (0, 1]. A value of 0.25 means we want a 1:4 ratio
                 of the minority to majority class after oversampling. We will create the a sampling dictionary using this ratio, with the keys corresponding to the class
                 and the values responding to the number of samples. Defaults to 0.25.
-            k_neighbors (int): The number of nearest neighbors to used to construct synthetic samples. Defaults to 5.
+            k_neighbors_default (int): The number of nearest neighbors to used to construct synthetic samples. This value will become smaller when there are less samples
```

Comment:
Typo: to used --> to use

Reply:
I think we could make this docstring more clear that this is the default used, but that the actual k_neighbors value used may be different if there are less samples? Currently a bit confusing to me or odd that it sounds like the parameter value will decrease :')